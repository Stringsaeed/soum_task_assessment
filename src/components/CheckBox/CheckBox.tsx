import { Check, Minus } from 'lucide-react-native';
import { StyleSheet } from 'react-native';
import Animated, { FadeInDown, FadeOutUp, LinearTransition } from 'react-native-reanimated';

import Box from '@/components/Box';

type Props = {
  status: 'checked' | 'unchecked' | 'indeterminate';
};

export default function CheckBox({ status }: Props) {
  return (
    <Box
      testID={`CheckBox-${status}`}
      layout={LinearTransition}
      style={styles.container}
      color={status !== 'unchecked' ? 'black' : 'white'}
      size={20}>
      {status === 'indeterminate' && (
        <Animated.View entering={FadeInDown} exiting={FadeOutUp}>
          <Minus color="white" size={16} />
        </Animated.View>
      )}
      {status === 'checked' && (
        <Animated.View entering={FadeInDown} exiting={FadeOutUp}>
          <Check color="white" size={16} />
        </Animated.View>
      )}
    </Box>
  );
}

const styles = StyleSheet.create({
  container: {
    borderColor: 'black',
    borderWidth: 2,
  },
});
