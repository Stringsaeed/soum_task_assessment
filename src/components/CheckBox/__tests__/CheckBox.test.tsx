import { render, screen } from '@testing-library/react-native';

import CheckBox from '../CheckBox';

describe('CheckBox', () => {
  it('should render', () => {
    render(<CheckBox status="unchecked" />);

    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('should handle indeterminate status', () => {
    render(<CheckBox status="indeterminate" />);

    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('should handle checked status', () => {
    render(<CheckBox status="checked" />);

    expect(screen.toJSON()).toMatchSnapshot();
  });
});
