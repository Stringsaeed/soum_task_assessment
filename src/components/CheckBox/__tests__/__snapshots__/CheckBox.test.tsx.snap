// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox should handle checked status 1`] = `
<View
  animatedStyle={
    {
      "value": {},
    }
  }
  collapsable={false}
  layout={[Function]}
  style={
    {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "aspectRatio": 1,
      "backgroundColor": "black",
      "borderColor": "black",
      "borderRadius": 8,
      "borderWidth": 2,
      "justifyContent": "center",
      "overflow": "hidden",
      "width": 20,
    }
  }
  testID="CheckBox-checked"
>
  <View
    animatedStyle={
      {
        "value": {},
      }
    }
    collapsable={false}
    entering={[Function]}
    exiting={[Function]}
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="16"
      bbWidth="16"
      fill="none"
      focusable={false}
      height={16}
      meetOrSlice={0}
      minX={0}
      minY={0}
      stroke="white"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth={2}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 16,
            "width": 16,
          },
        ]
      }
      vbHeight={24}
      vbWidth={24}
      width={16}
      xmlns="http://www.w3.org/2000/svg"
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeLinecap",
            "strokeLinejoin",
          ]
        }
        stroke={
          {
            "payload": 4294967295,
            "type": 0,
          }
        }
        strokeLinecap={1}
        strokeLinejoin={1}
        strokeWidth="2"
      >
        <RNSVGPath
          d="M20 6 9 17l-5-5"
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4294967295,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth="2"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;

exports[`CheckBox should handle indeterminate status 1`] = `
<View
  animatedStyle={
    {
      "value": {},
    }
  }
  collapsable={false}
  layout={[Function]}
  style={
    {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "aspectRatio": 1,
      "backgroundColor": "black",
      "borderColor": "black",
      "borderRadius": 8,
      "borderWidth": 2,
      "justifyContent": "center",
      "overflow": "hidden",
      "width": 20,
    }
  }
  testID="CheckBox-indeterminate"
>
  <View
    animatedStyle={
      {
        "value": {},
      }
    }
    collapsable={false}
    entering={[Function]}
    exiting={[Function]}
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="16"
      bbWidth="16"
      fill="none"
      focusable={false}
      height={16}
      meetOrSlice={0}
      minX={0}
      minY={0}
      stroke="white"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth={2}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 16,
            "width": 16,
          },
        ]
      }
      vbHeight={24}
      vbWidth={24}
      width={16}
      xmlns="http://www.w3.org/2000/svg"
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeLinecap",
            "strokeLinejoin",
          ]
        }
        stroke={
          {
            "payload": 4294967295,
            "type": 0,
          }
        }
        strokeLinecap={1}
        strokeLinejoin={1}
        strokeWidth="2"
      >
        <RNSVGPath
          d="M5 12h14"
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4294967295,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth="2"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;

exports[`CheckBox should render 1`] = `
<View
  animatedStyle={
    {
      "value": {},
    }
  }
  collapsable={false}
  layout={[Function]}
  style={
    {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "aspectRatio": 1,
      "backgroundColor": "white",
      "borderColor": "black",
      "borderRadius": 8,
      "borderWidth": 2,
      "justifyContent": "center",
      "overflow": "hidden",
      "width": 20,
    }
  }
  testID="CheckBox-unchecked"
/>
`;
